### PR TITLE
Rename *LogTX to *LogTreeTX and introduce a "treeless" LogTX

### PR DESCRIPTION
**Pull request by codingllama**
Transactions tied to a particular treeID now have "tree" in the name. The "new"
LogTX is acquired without a treeID and can only perform general queries. This
allows us to remove the hard-coded log #0 from several places.

### CODE REVIEW

**Comment by codingllama** [comment]
Issue #324

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,9 @@
-/trillian_log_server
-/trillian_map_server
+/.idea
```
Comment (AlCutter) [inline .gitignore]: ignore all ideas :)

File: storage/mysql/log_storage_test.go [hunk 2]
```diff
@@ -870,65 +865,215 @@ func TestLogRootUpdate(t *testing.T) {
...
+func runTestGetActiveLogIDsInternal(t *testing.T, test getActiveIDsTest, logID int64, wantIds []int64) {
+	s, err := NewLogStorage(DB)
+	if err != nil {
+		t.Fatalf("NewLogStorage() = (_, %v), want = (_, nil)", err)
+	}
 
+	logIDs, err := test.fn(s, context.TODO(), logID)
 	if err != nil {
-		t.Fatalf("Failed to get log ids: %v", err)
+		t.Errorf("%v = (_, %v_, want = (_, nil)", test.name, err)
```
Comment (AlCutter) [inline storage/mysql/log_storage_test.go]: missing a ")" after the returned values

File: storage/mysql/log_storage.go [hunk 3]
```diff
@@ -85,7 +85,69 @@ func (m *mySQLLogStorage) getDeleteUnsequencedStmt(num int) (*sql.Stmt, error) {
...
+
+// readOnlyLogTX implements storage.ReadOnlyLogTX
+type readOnlyLogTX struct {
+	tx *sql.Tx
+}
+
+func (m *mySQLLogStorage) Snapshot(ctx context.Context) (storage.ReadOnlyLogTX, error) {
+	tx, err := m.db.Begin()
+	if err != nil {
+		glog.Warningf("could not start ReadOnlyLogTX: %s", err)
```
Comment (AlCutter) [inline storage/mysql/log_storage.go]: initial capital is ok for log lines (just not errors) - same throughout

File: storage/mysql/log_storage_test.go (hunk 2 above)
Comment (codingllama) [inline storage/mysql/log_storage_test.go]: Done. Well spotted.

File: storage/mysql/log_storage.go (hunk 3 above)
Comment (codingllama) [inline storage/mysql/log_storage.go]: Done throughout.

**Comment by codingllama** [comment]
PTAL?

**Pull request merged**